hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/Gateway.java

```diff
@@ -70,6 +77,25 @@ public Void call() throws Exception {
     OzoneConfigurationHolder.setConfiguration(ozoneConfiguration);
     UserGroupInformation.setConfiguration(ozoneConfiguration);
     loginS3GUser(ozoneConfiguration);
+
+    if (StringUtils.isEmpty(ozoneConfiguration.get(
+            OzoneConfigKeys.OZONE_METADATA_DIRS))) {
+      //Setting ozone.metadata.dirs if not set so that server setup doesn't
+      // fail.
+      Path tmpMetaDir = Files.createTempDirectory(Paths.get(""),
+              "ozone_s3g_tmp_meta_dir");
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Please use `ShutdownHookManager` to add the shutdown hook.  Also, I think it should be executed after the other hook (which stops the server), so use a lower than default priority.

Reply:
@kerneltime Should I do the web ui change as part of another PR?